src/esm_runscripts/filelists.py

```diff
@@ -999,59 +1011,72 @@ def copy_files(config, filetypes, source, target):
-                                continue
-                            if os.path.isfile(file_target) and filecmp.cmp(
-                                file_source, file_target
-                            ):
-                                logger.debug(
-                                    f"Source and target file are identical, skipping {file_source}",
-                                )
-                                helpers.print_datetime(config)
-                                continue
-                            files_to_be_moved.append({
+                        if not os.path.isdir(dest_dir):
+                            # MA: ``os.makedirs`` creates the specified directory
+                            # and the parent directories if the last don't exist
+                            # (same as with ``mkdir -p <directory>>``)
+                            os.makedirs(dest_dir)
+                        if not os.path.isfile(file_source):
+                            logger.warning(f"File not found: {file_source}")
+                            helpers.print_datetime(config)
+                            missing_files.update({file_target: file_source})
+                            continue
+                        if os.path.isfile(file_target) and filecmp.cmp(
+                            file_source, file_target
+                        ):
+                            logger.debug(
+                                f"Source and target file are identical, skipping {file_source}",
+                            )
+                            helpers.print_datetime(config)
+                            continue
+                        files_to_be_moved.append(
+                            {
                                 "movement_method": movement_method,
                                 "file_source": file_source,
                                 "file_target": file_target,
-                            })
-
-                            # To avoid overwriting in general experiment folder
-                            if skip_intermediate == True:
-                                file_target = avoid_overwriting(
-                                    config, file_source, file_target
-                                )
+                            }
+                        )
 
-                            # Execute movement
-                            movement_method(file_source, file_target)
+                        # To avoid overwriting in general experiment folder
+                        if skip_intermediate == True:
+                            file_target = avoid_overwriting(
+                                config, file_source, file_target
+                            )
 
-                            successful_files.append(file_source)
-                        except IOError:
-                            logger.error(
-                                f"Could not copy {file_source} to {file_target} for unknown reasons.",
```

Comment:
I'm not sure I like that the `try/except` is totally removed. This will all go away when we do file dicts, but I would imagine that it tries to copy everything that it can (in parallel), gathers info about ones that fail, and then prints that info out at the end before an exit. Is that what is done in the implementation below (in `def movement`?)

Reply:
Yes, this is what is done. All the `ifs` here should not result in a crash, and if they do we want to be able to catch it and not just ignore it with a `try`. The `try` is valuable around the actual movement and that's what us done now in the function `movement`